src/redist/targets/GenerateLayout.targets

```diff
@@ -7,14 +7,19 @@
 
   <PropertyGroup>
     <!-- Blob storage directories are not stabilized, so these must refer to a package that does not stabilize -->
-    <AspNetCoreBlobVersion>$(MicrosoftAspNetCoreAppRuntimePackageVersion)</AspNetCoreBlobVersion>
-    <CoreSetupBlobVersion>$(MicrosoftNETCoreAppRuntimePackageVersion)</CoreSetupBlobVersion>
+    <!-- In source build, the layout does match, so use the runtime package versions rather than the VS redist versions -->
+    <AspNetCoreBlobVersion Condition=" '$(DotNetBuildFromSource)' == 'true' ">$(MicrosoftAspNetCoreAppRuntimePackageVersion)</AspNetCoreBlobVersion>
+    <AspNetCoreBlobVersion Condition=" '$(DotNetBuildFromSource)' != 'true' ">$(VSRedistCommonAspNetCoreSharedFrameworkx6470PackageVersion)</AspNetCoreBlobVersion>
+    <CoreSetupBlobVersion Condition=" '$(DotNetBuildFromSource)' == 'true' ">$(MicrosoftNETCoreAppRuntimePackageVersion)</CoreSetupBlobVersion>
+    <CoreSetupBlobVersion Condition=" '$(DotNetBuildFromSource)' != 'true' ">$(VSRedistCommonNetCoreSharedFrameworkx6470PackageVersion)</CoreSetupBlobVersion>
     <WindowsDesktopBlobVersion>$(VSRedistCommonWindowsDesktopSharedFrameworkx6470PackageVersion)</WindowsDesktopBlobVersion>
 
-    <!-- Change these individually to or $(CoreSetupBlobVersion), $(AspNetCoreBlobVersion), or appropriate fixed version depending if corresponding .Ref packages are unpinned. -->
-    <NETCoreAppTargetingPackBlobVersion>$(CoreSetupBlobVersion)</NETCoreAppTargetingPackBlobVersion>
-    <AspNetCoreTargetingPackBlobVersion>$(MicrosoftAspNetCoreAppRefInternalPackageVersion)</AspNetCoreTargetingPackBlobVersion>
-    <WindowsDesktopTargetingPackBlobVersion>$(WindowsDesktopBlobVersion)</WindowsDesktopTargetingPackBlobVersion>
+    <!-- Because the ref packs are not serviced in every build, the blob location may vary -->
+    <NETCoreAppTargetingPackBlobVersion Condition=" '$(DotNetBuildFromSource)' == 'true' ">$(CoreSetupBlobVersion)</NETCoreAppTargetingPackBlobVersion>
+    <NETCoreAppTargetingPackBlobVersion Condition=" '$(DotNetBuildFromSource)' != 'true' ">$(VSRedistCommonNetCoreTargetingPackx6470PackageVersion)</NETCoreAppTargetingPackBlobVersion>
+    <AspNetCoreTargetingPackBlobVersion Condition=" '$(DotNetBuildFromSource)' == 'true' ">$(MicrosoftAspNetCoreAppRuntimePackageVersion)</AspNetCoreTargetingPackBlobVersion>
```

Comment:
Did you mean to change this from `MicrosoftAspNetCoreAppRefInternalPackageVersion` to `MicrosoftAspNetCoreAppRuntimePackageVersion`?

Reply:
Based on your comment [above](https://github.com/dotnet/installer/pull/11936/files#r704795480), I think this means that if you're building the ref pack repo for a 6.0.2 runtime release, and the latest ref pack is 6.0.0, the directory layout would be `aspnetcore/Runtime/6.0.0/dotnet-aspnetcore-targeting-pack-6.0.0...tar.gz`, while the runtime itself would appear at: aspnetcore/Runtime/6.0.2/dotnet-aspnetcore-runtime-6.0.2.tar.gz`. Is that correct?

If that's right, then we should be using `MicrosoftAspNetCoreAppRefPackageVersion` and not the `InternalRefPackageVersion` here.

Reply:
We don't build the latest ref pack repo, we get the latest ref pack from SBRP.  I think this should keep using `MicrosoftAspNetCoreAppRefInternalPackageVersion`